File: isbnfinder/src/main/java/com/eplan/isbnfinder/validate/Isbn.java
package com.eplan.isbnfinder.validate;

/**
 * Represents an ISBN Number Whose Validity is Known
 * 
 * @author Alex Cohen
 *
 */
public class Isbn {
	
	public String isbn; // The ISBN Number
	public boolean validity; // Whether the ISBN is valid or not
	
	/**
	 * Creates a new ISBN Number
	 * 
	 * @param isbn The ISBN number as a String
	 * @param validity Whether the ISBN is valid or not
	 */
	public Isbn(String isbn, boolean validity) {
		this.isbn = isbn;
		this.validity = validity;
	}

	/**
	 * Gets the ISBN number as a String
	 * 
	 * @return The ISBN number
	 */
	public String getIsbn() {
		return this.isbn;
	}

	/**
	 * Sets the ISBN number
	 * 
	 * @param isbn The ISBN number to set
	 */
	public void setIsbn(String isbn) {
		this.isbn = isbn;
	}

	/**
	 * Checks if this ISBN is valid
	 * 
	 * @return true if valid, false if not
	 */
	public boolean isValid() {
		return this.validity;
	}

	/**
	 * Sets whether this ISBN is valid
	 * 
	 * @param validity true if valid, false if not
	 */
	public void setValidity(boolean validity) {
		this.validity = validity;
	}
	
	
}

File: isbnfinder/src/main/resources/frontend/angular-isbn-finder/src/app/home/isbn.ts
/**
 * Represents an ISBN number
 * 
 * @Author Alex Cohen
 */
export class Isbn {

    /**
     * The string representation of the ISBN number
     */
    public isbn: string;

    /**
     * Whether the ISBN number is valid
     */
    public validity: boolean;

    /**
     * Creates a new ISBN number with the given isbn string and a validity for whether the ISBN is valid
     * 
     * @param isbn The ISBN number as a string
     * @param validity True if the ISBN is valid, false if not
     */
    constructor(isbn: string, validity: boolean) {
        this.isbn = isbn;
        this.validity = validity;
    }

    /**
     * Gets the ISBN number
     * 
     * @returns The ISBN number as a string
     */
    public getIsbn(): string {
        return this.isbn;
    }

    /**
     * Sets the ISBN number
     * 
     * @param isbn The ISBN number to set 
     */
    public setIsbn(isbn: string) {
        this.isbn = isbn;
    }

    /**
     * Gets the validity
     * 
     * @returns True if the ISBN is valid, false if not
     */
    public getValidity(): boolean {
        return this.validity;
    }

    /**
     * Sets the validity
     * 
     * @param validity The validity to set
     */
    public setValidity(validity: boolean) {
        this.validity = validity;
    }

    /**
     * Checks if the other object passed in equals this ISBN object 
     * 
     * @param other The other object
     * @returns True if both objects are Isbn objects with the same isbn and validity. False otherwise.
     */
    public equals(other: any): boolean {
        // If other is an Isbn
        if (other instanceof Isbn) {
            let otherIsbn: Isbn = <Isbn>other; // Cast to Isbn

            // If it has the same isbn and validity, it's equal
            if (otherIsbn.getIsbn() === this.getIsbn() &&
                otherIsbn.getValidity() === this.getValidity()) {
                return true;
            } else { // If not, it isn't equal
                return false;
            }
        } else { // If other isn't an Isbn, it doesn't equal this Isbn
            return false;
        }
    }

}